pkg/controller/installation/bootstrapReconciler.go

```diff
@@ -158,6 +165,79 @@ func getConsoleRouteCR(ctx context.Context, serverClient k8sclient.Client) (*rou
 	return consoleRouteCR, nil
 }
 
+func (r *Reconciler) reconcilerGithubOauthSecret(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+
+	githubOauthSecretCR := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.ConfigManager.GetGHOauthClientsSecretName(),
+			Namespace: r.ConfigManager.GetOperatorNamespace(),
+		},
+	}
+
+	err := serverClient.Get(ctx, k8sclient.ObjectKey{Name: githubOauthSecretCR.Name, Namespace: githubOauthSecretCR.Namespace}, githubOauthSecretCR)
+	if !k8serr.IsNotFound(err) && err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	} else if k8serr.IsNotFound(err) {
+		githubOauthSecretCR.Data = map[string][]byte{
+			"clientId": []byte("dummy"),
+			"secret":   []byte("dummy"),
+		}
+	}
+	githubOauthSecretCR.ObjectMeta.ResourceVersion = ""
+
+	err = resources.CreateOrUpdate(ctx, serverClient, githubOauthSecretCR)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("Error reconciling Github OAuth secrets: %w", err)
+	}
+	logrus.Info("Bootstrap Github OAuth secrets successfully reconciled")
+
+	secretRoleCR := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.ConfigManager.GetGHOauthClientsSecretName(),
+			Namespace: r.ConfigManager.GetOperatorNamespace(),
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups:     []string{""},
+				Resources:     []string{"secrets"},
+				Verbs:         []string{"update", "get"},
+				ResourceNames: []string{r.ConfigManager.GetGHOauthClientsSecretName()},
+			},
+		},
+	}
+
+	err = resources.CreateOrUpdate(ctx, serverClient, secretRoleCR)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("Error creating Github OAuth secrets role: %w", err)
+	}
+
+	secretRoleBindingCR := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.ConfigManager.GetGHOauthClientsSecretName(),
+			Namespace: r.ConfigManager.GetOperatorNamespace(),
+		},
+		RoleRef: rbacv1.RoleRef{
+			Name: secretRoleCR.GetName(),
+			Kind: "Role",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Name: "dedicated-admins",
+				Kind: "Group",
+			},
+		},
+	}
+
+	err = resources.CreateOrUpdate(ctx, serverClient, secretRoleBindingCR)
```

Comment:
Same as above - it would be nicer to use `controllerutil.CreateOrUpdate` here. :)